ci/container_index/lib/checks/schema_validation.py

```diff
@@ -0,0 +1,141 @@
+from ci.container_index.lib.checks.basevalidation\
+    import Validator, BasicSchemaValidator, StringFieldValidator
+import ci.container_index.lib.utils as index_utils
+from ci.container_index.lib.constants import *
+from uuid import uuid4
+from ci.container_index.lib.utils import\
+    load_yaml, dump_yaml
+
+
+class TopLevelProjectsValidator(Validator):
+
+    def __init__(self, validation_data, file_name):
+        super(TopLevelProjectsValidator, self).__init__(
+            validation_data, file_name)
+
+    def _perform_validation(self):
+        self.message.title = "Top level projects"
+        if not self.validation_data.get(
+            FieldKeys.PROJECTS
+        ):
+            self._invalidate(
+                "Index data should begin with "
+                "top level \"Projects:\""
+            )
+            return
+        if not isinstance(
+            self.validation_data.get(FieldKeys.PROJECTS),
+            list
+        ):
+            self._invalidate(
+                "Projects must be provided as a list of dicts."
+            )
+            return
+
+
+class IDValidator(BasicSchemaValidator):
+
+    def __init__(self, validation_data, file_name):
+        super(IDValidator, self).__init__(
+            validation_data, file_name)
+        self.field_name = FieldKeys.ID
+
+    def _extra_validation(self):
+        self.message.title = "Id field validation"
+        if not isinstance(
+            self.validation_data.get(FieldKeys.ID), int
+        ):
+            self._invalidate("id field must be an integer.")
+            return
+
+
+class AppIDValidator(StringFieldValidator):
+
+    def __init__(self, validation_data, file_name):
+        super(AppIDValidator, self).__init__(
+            validation_data, file_name)
+        self.field_name = FieldKeys.APP_ID
+        self.message.title = "App ID Validation"
+
+    def _extra_validation_1(self):
+        if (self.validation_data.get(self.field_name) !=
+           self.file_base_name.split(".")[0]):
+            self._invalidate(
+                str.format(
+                    "{} must be the same as the file name",
```

Comment:
provide the filename as well to be explicit.

Reply:
Fixed